### ISSUE TITLE
@RegisterExtension field is silently ignored if null at evaluation time

### ISSUE THREAD

**Issue by sbrannen** [junit-team/junit5#1727]
## Overview

If a field in a test class (either static or non-static) is annotated with `@RegisterExtension` but is never assigned a non-null value before evaluation for extension registration purposes, that situation is currently silently ignored thereby masking the user configuration error.

For example, the following test case passes without any error or warning.

```java
class RegisterExtensionWithNullFieldTestCase {

	@RegisterExtension
	BeforeEachCallback beforeEach;

	@Test
	void test() {
	}

}
```

## Deliverables

- [x] Fail the corresponding test class or test method with an exception if a `@RegisterExtension` field is `null` at evaluation time.

**Comment by sbrannen** [comment]
**Update**:

The Javadoc for `@RegisterExtension` actually states the following:

> `@RegisterExtension` fields must not be `private` or `null` (when evaluated) but may be either `static` or non-static.

The _intention_ that `null` is not allowed at evaluation time was therefore always present. 😉

**Comment by sbrannen** [comment]
This has been fixed in `master` in c2d3112a0ab593fccfd8649170b6e66d3aa7e116.
